src/BuildScriptGeneratorCli/Options/BuildScriptGeneratorOptionsSetup.cs

```diff
@@ -0,0 +1,26 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.BuildScriptGenerator;
+
+namespace Microsoft.Oryx.BuildScriptGeneratorCli.Options
+{
+    public class BuildScriptGeneratorOptionsSetup : IConfigureOptions<BuildScriptGeneratorOptions>
+    {
+        private readonly IEnvironment _environment;
+
+        public BuildScriptGeneratorOptionsSetup(IEnvironment environment)
```

Comment:
FYI: Ideal way is to get `IConfiguration` from DI and use it to get values of keys (for example: `configuration.GetValue("ENABLE_DYNAMIC_INSTALL")`) and then bind to different properties of this options. The advantage of this way is that `configuration.GetValue` call would get the hierarchically resolved value to you i.e command line value over environment variable over file settings

What I have here is a stop-gap solution to prevent this PR to get very big. I will later send out a different PR as it is going to have substantial changes.